<a href="https://colab.research.google.com/github/damianmtz98/PCD_MCD/blob/main/P2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Analisis inicial de los datos.**

In [11]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [12]:
import warnings
warnings.filterwarnings('ignore')

In [13]:
df_fraud = pd.read_csv('Fraud.csv')
df_fraud.head(10)

TYPE    AMOUNT    ID_ORIGIN  OLD_BALANCE_ORIG  NEW_BALANCE_ORIG  \
0   PAYMENT   9839.64  C1231006815         170136.00         160296.36   
1   PAYMENT   1864.28  C1666544295          21249.00          19384.72   
2  TRANSFER    181.00  C1305486145            181.00              0.00   
3  CASH_OUT    181.00   C840083671            181.00              0.00   
4   PAYMENT  11668.14  C2048537720          41554.00          29885.86   
5   PAYMENT   7817.71    C90045638          53860.00          46042.29   
6   PAYMENT   7107.77   C154988899         183195.00         176087.23   
7   PAYMENT   7861.64  C1912850431         176087.23         168225.59   
8   PAYMENT   4024.36  C1265012928           2671.00              0.00   
9     DEBIT   5337.77   C712410124          41720.00          36382.23   

       ID_DEST  OLD_BALANCE_DEST  NEW_BALANCE_DEST  FRAUD  is_FLAGGED_FRAUD  
0  M1979787155               0.0              0.00      0                 0  
1  M2044282225               0.0              0.00      0                 0  
2   C553264065               0.0              0.00      1                 0  
3    C38997010           21182.0              0.00      1                 0  
4  M1230701703               0.0              0.00      0                 0  
5   M573487274               0.0              0.00      0                 0  
6   M408069119               0.0              0.00      0                 0  
7   M633326333               0.0              0.00      0                 0  
8  M1176932104               0.0              0.00      0                 0  
9   C195600860           41898.0          40348.79      0                 0

* Tipo de movimiento (TYPE): CASH-IN, CASH-OUT, DEBIT, PAYMENT and TRANSFER.
* Monto de la transacción (AMOUNT):  Monto de la transacción en moneda local.
* Id del cliente inicial (ID_ORIGIN): Cliente que inició la transacción.
* Balance inicial (OLD_BALANCE_ORIG): Saldo inicial antes de la transacción.
* Balance final (NEW_BALANCE_ORIGINAL): Nuevo saldo después de la transacción.
* Id del cliente destino (ID_DEST): cliente que es el destinatario de la transacción.
* Balance inicial del destinatario (OLD_BALANCE_DEST): Destinatario del saldo inicial antes de la transacción. Tenga en cuenta que no hay información para los clientes que comienzan con M (Comerciantes).
* Balance final del destinatario (NEW_BALANCE_DEST): Destinatario del nuevo saldo después de la transacción. Tenga en cuenta que no hay información para los clientes que comienzan con M (Comerciantes).
* ***Si es fraude o no (FRAUD): son las transacciones realizadas por los agentes fraudulentos dentro de la simulación. ***
* Si esta marcado como fraude en el sistema (isFlaggedFraud)

En este conjunto de datos específico, el comportamiento fraudulento de los agentes tiene como objetivo obtener ganancias tomando el control de las cuentas de los clientes e intentar vaciar los fondos transfiriéndolos a otra cuenta y luego cobrándolos del sistema.


FRAUD es nuestra variable objetivo.





In [17]:
df_fraud.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10070 entries, 0 to 10069
Data columns (total 10 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   TYPE              10070 non-null  object 
 1   AMOUNT            10070 non-null  float64
 2   ID_ORIGIN         10070 non-null  object 
 3   OLD_BALANCE_ORIG  10070 non-null  float64
 4   NEW_BALANCE_ORIG  10070 non-null  float64
 5   ID_DEST           10070 non-null  object 
 6   OLD_BALANCE_DEST  10070 non-null  float64
 7   NEW_BALANCE_DEST  10070 non-null  float64
 8   FRAUD             10070 non-null  int64  
 9   is_FLAGGED_FRAUD  10070 non-null  int64  
dtypes: float64(5), int64(2), object(3)
memory usage: 786.8+ KB


In [18]:
df_fraud.shape

(10070, 10)

Contamos con valores flotantes y de tipo objeto y tenemos exactamente 10070 datos y 10 columnas en total.


In [23]:
df_fraud.describe()


AMOUNT  OLD_BALANCE_ORIG  NEW_BALANCE_ORIG  OLD_BALANCE_DEST  \
count  1.007000e+04      1.007000e+04      1.007000e+04      1.007000e+04   
mean   1.042986e+05      8.881769e+05      9.092621e+05      9.332923e+05   
std    2.692014e+05      2.129458e+06      2.175124e+06      2.677966e+06   
min    2.390000e+00      0.000000e+00      0.000000e+00      0.000000e+00   
25%    4.397795e+03      1.292500e+02      0.000000e+00      0.000000e+00   
50%    1.283686e+04      2.136487e+04      1.024826e+04      0.000000e+00   
75%    1.143752e+05      1.748616e+05      1.724218e+05      2.819345e+05   
max    1.000000e+07      1.290000e+07      1.300000e+07      1.950000e+07   

       NEW_BALANCE_DEST         FRAUD  is_FLAGGED_FRAUD  
count      1.007000e+04  10070.000000           10070.0  
mean       1.103151e+06      0.006753               0.0  
std        3.034778e+06      0.081901               0.0  
min        0.000000e+00      0.000000               0.0  
25%        0.000000e+00      0.000000               0.0  
50%        0.000000e+00      0.000000               0.0  
75%        2.520552e+05      0.000000               0.0  
max        2.260000e+07      1.000000               0.0

Descripción de los datos

In [22]:
df_fraud.duplicated().any()

False

No contamos con datos duplicados.